plugins/modules/win_pssession_configuration.ps1

```diff
@@ -0,0 +1,551 @@
+}
+
+$opt_pssc       = ConvertFrom-AnsibleOptions -OptionSet $pssc_options
+$opt_session    = ConvertFrom-AnsibleOptions -OptionSet $session_configuration_options
+
+$existing = Get-PSSessionConfiguration -Name $opt_session.Name -ErrorAction SilentlyContinue
+
+try {
+    if ($opt_pssc.Count) {
+        # config file options were passed to the module, so generate a config file from those
+        $desired_config = Write-GeneratedSessionConfiguration -ParameterSet $opt_pssc
+    }
+    if ($existing) {
+        # the endpoint is registered
+        if ($existing.ConfigFilePath -and (Test-Path -LiteralPath $existing.ConfigFilePath)) {
+            # the registered endpoint uses a config file
+            if ($desired_config) {
+                # a desired config file exists, so compare it to the existing one
+                $content_match = $existing |
+                    Compare-ConfigFile -NewConfigFile $desired_config -Params $opt_pssc -UseExistingIfMissing (
+                        $module.Params.lenient_config_fields | ConvertFrom-SnakeCase
+                    )
+            }
+            else {
+                # existing endpoint has a config file but no config file options were passed, so there is no match
+                $content_match = $false
+            }
+        }
+        else {
+            # existing endpoint doesn't use a config file, so it's a match if there are also no config options passed
+            $content_match = $opt_pssc.Count -eq 0
+        }
+
+        $session_match = Compare-SessionOption -DesiredOptions $opt_session -ExistingOptions $existing
+    }
+
+    $state = $module.Params.state
+
+    $create = $state -eq 'present' -and (-not $existing -or -not $content_match)
+    $remove = $existing -and ($state -eq 'absent' -or -not $content_match)
+    $session_change = -not $session_match -and $state -ne 'absent'
+
+    $module.Result.changed = $create -or $remove -or $session_change
+
+    # In this module, we pre-emptively remove the session configuratin if there's any change
+    # in the config file options, and then re-register later if needed.
+    # But if the RunAs credential is wrong, the register will fail, and since we already removed
+    # the existing one, it will be gone.
+    #
+    # So let's ensure we can actually use the credential by running something (whoami.exe) with the become
+    # module util, that way we can fail before touching the existing config.
+    if ($opt_session.Contains('RunAsCredential')) {
+        $cred = $opt_session.RunAsCredential
+        try {
+            $result = [Ansible.Become.BecomeUtil]::CreateProcessAsUser($cred.Username, $cred.GetNetworkCredential().Password, 'whoami.exe')
+            $module.Debug("RunAsCredential check whoami.exe output: $($result.StandardOut)")
+            if ($result.ExitCode -ne 0) {
+                throw
+            }
+        }
+        catch {
+            $module.FailJson('Could not validate RunAs Credential.')
```

Comment:
It would be good to include some error information for both the user's and developers perspective

```suggestion
            $module.FailJson('Could not validate RunAs Credential: $($_.Exception.Message)', $_)
```